actors/runtime/src/main/java/cloud/orbit/actors/runtime/Hosting.java

```diff
@@ -376,25 +376,23 @@ public void actorDeactivated(RemoteReference remoteReference)
             nodeAddress = clusterPeer.localAddress();
         }
 
-        // Do we have a target node yet?
-        if (nodeAddress == null)
-        {
-            // If not, select randomly
-            nodeAddress = await(selectNode(interfaceClass.getName()));
-        }
+        // Do we have a target node yet? If not, select randomly
+        Task<NodeAddress> nodeAddressTask = nodeAddress != null ? Task.fromValue(nodeAddress) : selectNode(interfaceClass.getName());
 
-        // Push our selection to the distributed cache (if possible)
-        NodeAddress otherNodeAddress = distributedDirectory.putIfAbsent(remoteKey, nodeAddress);
+        return nodeAddressTask.thenApply((nodeAddress1) -> {
+            // Push our selection to the distributed cache (if possible)
+            NodeAddress otherNodeAddress = distributedDirectory.putIfAbsent(remoteKey, nodeAddress1);
 
-        // Someone else beat us to placement, use that node
-        if (otherNodeAddress != null)
-        {
-            nodeAddress = otherNodeAddress;
-        }
+            // Someone else beat us to placement, use that node
+            if (otherNodeAddress != null)
+            {
+                nodeAddress1 = otherNodeAddress;
```

Comment:
Can we come up with a better name than nodeAddress1? What is this actually suppose to represent? If we already have nodeAddress earlier in the method it would be nice to understand the distinction.